packages/ibm-products-web-components/src/components/truncated-text/truncated-text.ts

```diff
@@ -0,0 +1,214 @@
+/**
+ * @license
+ *
+ * Copyright IBM Corp. 2025, 2025
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+import { html, LitElement } from 'lit';
+import { property, query, state } from 'lit/decorators.js';
+import { classMap } from 'lit/directives/class-map.js';
+import { carbonElement as customElement } from '@carbon/web-components/es/globals/decorators/carbon-element.js';
+import { prefix, carbonPrefix } from '../../globals/settings';
+import '@carbon/web-components/es/components/tooltip/index.js';
+import '@carbon/web-components/es/components/button/button.js';
+import '@carbon/web-components/es/components/link/index.js';
+
+import styles from './truncated-text.scss?lit';
+
+const componentName = 'truncated-text';
+export const blockClass = `${prefix}--${componentName}`;
+const elementName = `${prefix}-${componentName}`; // c4p-truncated-text
+
+/**
+ * TruncatedText.
+ *
+ * @element c4p-truncated-text
+ */
+@customElement(elementName)
+export class CDSTruncatedText extends LitElement {
+  /**
+   * The maximum number of lines to display before truncation.
+   */
+  @property({ type: Number, reflect: true }) lines = 0;
+
+  /**
+   * The string value to be truncated.
+   */
+  @property({ type: String, attribute: 'value', reflect: true }) value = '';
+
+  /**
+   * The label on expand button.
+   */
+  @property({ attribute: 'expand-label', type: String, reflect: true })
+  expandLabel = '...more';
+
+  /**
+   * The label on the collapse button.
+   */
+  @property({ attribute: 'collapse-label', type: String, reflect: true })
+  collapseLabel = '...less';
```

Comment:
I recommend we stick with “View” or “Read” as our defaults?

```suggestion
  @property({ attribute: 'expand-label', type: String, reflect: true })
  expandLabel = 'View more';

  /**
   * The label on the collapse button.
   */
  @property({ attribute: 'collapse-label', type: String, reflect: true })
  collapseLabel = 'View less';
```